unoconv

```diff
@@ -197,23 +197,47 @@ def find_offices():
 #        if not glob.glob(realpath(basepath, basis, program, 'python-core-*')):
 #            continue
 
-        for pythonbinary in pythonbinaries:
-            if os.path.isfile(realpath(unopath, pythonbinary)):
-                info(3, "Found %s in %s" % (pythonbinary, unopath))
-                ret.append(Office(basepath, urepath, unopath, officelibrary, officebinary,
-                                  realpath(unopath, pythonbinary), pythonhome))
+
+        ### Find suitable Python executable
+        pythonpath_candidates = [
+
+            # Regular unopath
+            unopath,
+
+            # LibreOffice 5.4.6.2 on Mac OS X 10.13.3 ships
+            # the "python" executable in the "Resources" folder.
+            realpath(basepath, 'Resources'),
+        ]
+
+        python_effective = find_file_candidate(pythonpath_candidates, pythonbinaries)
+
+        if python_effective:
+            info(3, "Found Python at %s" % python_effective)
+            office = Office(basepath, urepath, unopath, officelibrary, officebinary,
+                            python_effective, pythonhome)
         else:
             info(3, "Considering %s" % basepath)
-            ret.append(Office(basepath, urepath, unopath, officelibrary, officebinary,
-                              sys.executable, None))
+            office = Office(basepath, urepath, unopath, officelibrary, officebinary,
+                            sys.executable, None)
+
+        ret.append(office)
+
     return ret
 
+def find_file_candidate(folders, filenames):
+    for folder in folders:
+        for filename in filenames:
+            candidate = realpath(folder, filename)
+            if os.path.isfile(candidate):
+                return candidate
```

Comment:
Shouldn't this function return a list of all that are found, not just one?

Reply:
In this case, i really wanted the helper function to return the single outcome from a list of possible candidates. I wouldn't know what to do with multiple outcomes on the caller side. Who would decide which one to use?

So, maybe we should rename the function to `find_file_effective` or `find_file_from_candidates` then. Would that help?

Reply:
`find_python` or `find_executable` seems better. "file_effective" doesn't explain what file or why it's effective.

Reply:
Other than that it looks good.

Reply:
It's called `find_executable` now. Hope this helps and thanks again.